stonesoup/movable/action/move_position_action.py

```diff
@@ -104,3 +104,105 @@ def __iter__(self):
                     yield MovePositionAction(generator=self,
                                              end_time=self.end_time,
                                              target_value=target_value)
+
+
+class SamplePositionActionGenerator(ActionGenerator):
+    """Base action generator for sampling approaches to action generation. The action
+    generator requires the user to define a number of samples to generate
+    (:attr:`n_samples`) according to the defined sampling technique."""
+
+    action_space: np.ndarray = Property(
+        default=None,
+        doc="The bounds of the action space that should not be exceeded. Of shape (ndim, 2) "
+            "where ndim is the length of the action_mapping. For example, "
+            ":code:`np.array([[xmin, xmax], [ymin, ymax]])`."
+    )
+
+    action_mapping: Sequence[int] = Property(
+        default=(0, 1),
+        doc="The state dimensions that actions are applied to."
+    )
+
+    n_samples: int = Property(
+        default=10,
+        doc="Number of samples to generate. This does not include the action "
+        "to remain at the current position, meaning :attr:`n_samples` +1 "
+        "actions will be generated."
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.action_space is not None:
+            if len(self.action_space) != len(self.action_mapping):
+                raise ValueError(f"Dimensions of action_space {self.action_space.shape} "
+                                 f"are not compatible with action_mapping of length "
+                                 f"{len(self.action_mapping)}. action_space should be "
+                                 f"of shape (ndim, 2) where ndim is the length of the "
+                                 f"action_mapping.")
+
+            if (np.any(self.current_value[self.action_mapping, :] < self.action_space[:, [0]])
+                    or np.any(self.current_value[self.action_mapping, :] > self.action_space[:, [1]])):  # noqa: E501
+                raise ValueError(f"Initial platform location {self.current_value} is not within "
+                                 f"the bounds of the action space {self.action_space}.")
+
+    @property
+    def default_action(self):
+        return MovePositionAction(generator=self,
+                                  end_time=self.end_time,
+                                  target_value=self.current_value)
+
+    def __contains__(self, item):
+        return item in iter(self)
+
+    @abstractmethod
+    def __iter__(self):
+        raise NotImplementedError
+
+
+class CircleSamplePositionActionGenerator(SamplePositionActionGenerator):
```

Comment:
suggest to import these classes in this file in the __init__.py to allow for neater imports elsewhere.